servers/cu/src/domain/lib/evaluate.js

```diff
@@ -83,6 +83,7 @@ export function evaluateWith (env) {
 
   const saveLatestProcessMemory = saveLatestProcessMemorySchema.implement(env.saveLatestProcessMemory)
 
+  let totalGasUsed = BigInt(0)
```

Comment:
As it is, `totalGasUsed` would continue to accumulate across **_all evaluation streams, across all processes_**, eventually causing every eval stream to produce a checkpoint. Probably not what we want.

Instead this variable need to be moved into the function below, so that it is scoped to the single evaluation stream, then garbage collected. Perhaps within the function starting on ln.90

Reply:
🤦